File: edition/2017/javascript/Sorting_algorithms/bubble_sort.js
function bubbleSort(items) 
{
    var length = items.length;
    for (var i = 0; i < length; i++) 
    {
        for (var j = 0; j < (length - i - 1); j++) 
        { 
            if(items[j] > items[j+1]) 
            {
                var tmp = items[j];  
                items[j] = items[j+1]; 
                items[j+1] = tmp; 
            }
        } 
    }
}

File: edition/2017/ruby/Sorting_algorithms/bubble_sort.rb
def bubble_sort(array)
    return array if array.size <= 1
    swap = true
    while swap
        swap = false
        (array.length - 1).times do |x|
            if array[x] > array[x+1]
                array[x], array[x+1] = array[x+1], array[x]
                swap = true
            end
        end
    end
    array
end